### ISSUE TITLE
Router: Missing isNavigating property

### ISSUE THREAD

**Issue by Vheissu** [aurelia/aurelia#1340]
In many valid scenarios, you want to know when the router is doing somethnig. In v1 we had a property that allowed us to check for this, we also had access to events emitted throughout the routing lifecycle via the Event Aggregator. In v2, we have no such property nor events to determine what the router is doing. The closest we have to this are lifecycle hooks. Where lifecycle hooks do not work are for loading indicators especially. In Aurelia v1 router we had numerous router states, including a isNavigating property.

**Comment by Sayan751** [comment]
@Vheissu The events are there. Those are as follows:

```
au:router:location-change
au:router:navigation-start
au:router:navigation-end
au:router:navigation-cancel
au:router:navigation-error
```

You may choose to subscribe to those the classical way, using the `EventAggregator`. There is another alternative of using the new `IRouterEvents#subscribe`. The advantage is that using that the allowed event names and corresponding strongly typed events will be available in IDE IntelliSense. 

![2022-03-05_08h59_25](https://user-images.githubusercontent.com/3128881/156874479-fd9240e1-5d41-40cc-83b4-7e15c4589645.png)
![2022-03-05_09h00_33](https://user-images.githubusercontent.com/3128881/156874481-b7d8e38a-10f9-437a-8389-75ab6cc65b34.png)
![2022-03-05_09h01_23](https://user-images.githubusercontent.com/3128881/156874482-85239447-9136-437e-b3f4-2207b56f8253.png)
![image](https://user-images.githubusercontent.com/3128881/156874552-38d23401-b8c3-478b-8bb4-44c37c511c0d.png)

Other than the router events, I never personally used the `isNavigating` property in a real app, even though that is prevalent in the toy example apps or tutorials. I think all the real app complexities can be served well by the granular events already. Therefore, I am inclined towards not reintroducing that property unless we heard about real use-cases. @bigopon need your feedback here; what do think about this?

In summary, I think that the actionable item for this issue would be to simply update the docs. Do let me know if we have an agreement, and I will reassign the issue back to you @Vheissu.
